core/runtime/src/main/java/io/quarkus/runtime/util/ClassPathUtils.java

```diff
@@ -200,6 +204,16 @@ public static void consumeStream(URL url, Consumer<InputStream> consumer) throws
         }
     }
 
+    private static URI toURI(URL url) throws IOException {
+        final URI uri;
+        try {
+            uri = new URI(url.toString());
```

Comment:
```suggestion
            uri = url.toURI();
```

Reply:
I'm not sure about the extra checks that `toURI()` does.